### DISCUSSION TITLE
How are passive and active orientation conventions defined in Neper?

### DISCUSSION THREAD

**Discussion by jeanmichelscherer** [neperfepx/neper#641] [Q&A] [answered]
Hi,

I am wondering whether Neper defines the passive rotation convention as the rotation that transforms a vector in the specimen frame to a vector in the crystal frame (as done [here]( https://pymicro.readthedocs.io/projects/pymicro/en/latest/cookbook/euler_angles.html) for instance). If that is the intended behavior, then I do not understand the result below:

- I generate a single orientation such that the [001] specimen axis coincides with the [111] crystal axis:
```
neper -T -n 1 -ori 'fiber(1,1,1,0,0,1)' -oridescriptor 'euler-bunge:passive' -format 'ori'
```

- The `n1-id1.ori` file contains the following angles:
`135.000000000000   54.735610317245 -172.197179036517`

- Then I run the following python script to check whether the transpose of the passive rotation matrix transforms the [111] crystal direction back to the [001] specimen direction as expected:
```
import numpy as np
import matplotlib.pyplot as plt

def rotation(phi1,Phi,phi2):
    # Create the rotation matrix associated to Bunge-Euler angles (passive convention)
    R = np.array([[np.cos(phi1)*np.cos(phi2)-np.sin(phi1)*np.sin(phi2)*np.cos(Phi),
                np.sin(phi1)*np.cos(phi2)+np.cos(phi1)*np.sin(phi2)*np.cos(Phi),
                np.sin(phi2)*np.sin(Phi)],
                [-np.cos(phi1)*np.sin(phi2)-np.sin(phi1)*np.cos(phi2)*np.cos(Phi),
                -np.sin(phi1)*np.sin(phi2)+np.cos(phi1)*np.cos(phi2)*np.cos(Phi),
                np.cos(phi2)*np.sin(Phi)],
                [np.sin(phi1)*np.sin(Phi),
                -np.cos(phi1)*np.sin(Phi),
                np.cos(Phi)]])
    return R

# Load Bunge-Euler angles
deg2rad = np.pi/180.
phi1, Phi, phi2 = np.loadtxt('n1-id1.ori')*deg2rad

# Define the passive rotation matrix
R = rotation(phi1,Phi,phi2)

# Rotate (111) crystal direction to specimen frame (Not OK)
Z1 = np.dot(R.transpose(),np.array([1,1,1])/np.sqrt(3))
# 
Z2 = np.dot(R,np.array([1,1,1])/np.sqrt(3))

fig = plt.figure()
ax = fig.add_subplot(111, projection='3d')

ax.quiver(0.,0.,0.,Z1[0],Z1[1],Z1[2],color='blue')
ax.quiver(0.,0.,0.,Z2[0],Z2[1],Z2[2],color='red')

ax.set_xlim([-1., 1.])
ax.set_ylim([-1., 1.])
ax.set_zlim([-1., 1.])
ax.set_xlabel('X')
ax.set_ylabel('Y')
ax.set_zlabel('Z')
plt.show()
```

- I would expect the blue arrow to be aligned with the Z axis, but instead, the red arrow is.

Am I missing something here?

- On a related topic: How are the elements of the rotation matrix ordered (row-wise or column-wise) in the `.ori` file when the `rotmat` `oridescriptor` is used? I could not find that info in the documentation.

Thanks,
Jean-Michel

**Comment by mkasemer** [comment, marked as answer]
@jeanmichelscherer : The usual terminology, as far as I am aware, is that passive indicates a basis transformation from the crystal basis to the sample basis (crystal-to-sample), and that active indicates a basis transformation from the sample basis to the crystal basis (sample-to-crystal). This is what we have implemented in FEPX, and what is described in the documentation there (see: [here](https://fepx.info/doc/input.html?highlight=passive)).

Assuming this, do your results make sense?

**Reply by jeanmichelscherer** [reply]
If the passive convention is defined as such in Neper/FEPX, then the results I get make sense. I think it could be useful to add this info to the Neper documentation as well.

I found the following references that define the passive convention (differently) as the transformation from sample basis to crystal basis:
- [MOOSE framework](https://mooseframework.inl.gov/source/materials/crystal_plasticity/ComputeElasticityTensorConstantRotationCP.html)
- [NEML](https://neml.readthedocs.io/en/dev/crystal_plasticity.html)
- https://www.nature.com/articles/s41597-022-01525-w

Jean-Michel

**Reply by mkasemer** [reply]
**EDIT** I suspect that you may be correct. I will cede to @rquey here.
